### PR TITLE
Split `DATABRICKS_JDBC_URL` into fine-grained components 

### PR DESCRIPTION
**Pull request by findinpath**
### Delta Lake product tests infrastructure

Use the following value for the Databricks JDBC URL:

```
jdbc:databricks://${DATABRICKS_SERVER_HOSTNAME}:443/default;transportMode=http;ssl=1;httpPath=sql/protocolv1/o/${DATABRICKS_ORGANISATION_ID}/${DATABRICKS_CLUSTER_ID};AuthMech=3;EnableArrow=0
```

in order to provide more transparence on the settings used
for connecting via JDBC to the Databricks compute cluster.

This change requires the introduction of new secrets in the CI:

- `DATABRICKS_SERVER_HOSTNAME`
- `DATABRICKS_ORGANISATION_ID`
- `DATABRICKS_73_CLUSTER_ID`
- `DATABRICKS_91_CLUSTER_ID`
- `DATABRICKS_104_CLUSTER_ID`

In case this change is being accepted, please add a follow-up task to remove in a few weeks the following secrets:

- `DATABRICKS_73_JDBC_URL`
- `DATABRICKS_91_JDBC_URL`
- `DATABRICKS_104_JDBC_URL`

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/AbstractSinglenodeDeltaLakeDatabricks.java [hunk 1]
```diff
@@ -46,18 +46,17 @@ public AbstractSinglenodeDeltaLakeDatabricks(Standard standard, DockerFiles dock
     @Override
     public void extendEnvironment(Environment.Builder builder)
     {
-        String databricksTestJdbcUrl = databricksTestJdbcUrl();
+        String databricksClusterId = databricksClusterId();
+        String databricksServerHostname = requireNonNull(System.getenv("DATABRICKS_SERVER_HOSTNAME"), "Environment DATABRICKS_LOGIN was not set");
+        String databricksOrganisationId = requireNonNull(System.getenv("DATABRICKS_ORGANISATION_ID"), "Environment DATABRICKS_LOGIN was not set");
```
Comment (ebyhr) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/AbstractSinglenodeDeltaLakeDatabricks.java:51]: nit: Please fix error message.
```suggestion
        String databricksServerHostname = requireNonNull(System.getenv("DATABRICKS_SERVER_HOSTNAME"), "Environment DATABRICKS_SERVER_HOSTNAME was not set");
        String databricksOrganisationId = requireNonNull(System.getenv("DATABRICKS_ORGANISATION_ID"), "Environment DATABRICKS_ORGANISATION_ID was not set");
```

**Comment by findepi** [comment]
I wouldn't bother doing that.  The more secrets the more overhead.
It should be fine to just document how DATABRICKS_JDBC_URL is constructed (as a code comment in ci.yml, where the secret is being used).

**Comment by findinpath** [comment]
I would argue that this change brings more transparency about how the Databricks JDBC URL looks like for users which don't have access to Trino's Databricks runtime. In consequence, they are able to test easier their changes, changes related to introducing new features on top of Delta Lake.

**Comment by findepi** [comment]
That's why we need to know how DATABRICKS_JDBC_URL should be constructed.
This can be achieved by replacing a secret with 5 others, or via a code comment.
I am for the code comment. 

Also, newer DBR version may require slightly different URL, so would prefer to keep it as a single secret that's much easier to manage.

**Pull request closed without merging**